runtime/lib/binary.cpp

```diff
@@ -4,6 +4,7 @@
 
 #include <fstream>
 
+#pragma clang diagnostic ignored "-Wcovered-switch-default"
```

Comment:
Can you use `#pragma clang diagnostic push` and `#pragma clang diagnostic pop` to surround only the impacted header?  We only want to ignore warnings for external source.